src/fontra/client/web-components/range-slider.js

```diff
@@ -0,0 +1,273 @@
+      background: #282828;
+      border: none;
+      cursor: pointer;
+    }
+
+    .slider::-ms-track {
+      height: 5px;
+      background: dimgray;
+    }
+
+    .range-slider-options {
+      position: relative;
+      display: flex;
+      justify-content: space-between;
+      z-index: -1;
+    }
+
+    .range-slider-options > span {
+      display: block;
+      width: 2px;
+      height: 3.5px;
+      opacity: 0.65;
+      background: dimgray;
+    }
+  `;
+
+  static properties = {
+    name: { type: String },
+    minValue: { type: Number },
+    maxValue: { type: Number },
+    defaultValue: { state: true },
+    tickMarksPositions: { type: Array },
+    step: { type: Number },
+    onChangeCallback: { type: Function },
+  };
+
+  constructor() {
+    super();
+    // Fallbacks for attributes that are not defined when calling the component
+    this.name = "Slider";
+    this.minValue = 0;
+    this.maxValue = 100;
+    this.defaultValue = this.midValue;
+    this.tickMarksPositions = [];
+    this.step = 0.1;
+    this.onChangeCallback = () => {};
+  }
+
+  render() {
+    return html`
+      <div class="wrapper">
+        <div class="numeric-input">
+          <section class="slider-input">
+            <input
+              type="number"
+              @input=${this.changeValue}
+              class="slider-default-value"
+              min=${this.minValue}
+              max=${this.maxValue}
+              step=${this.step}
+              pattern="[0-9]+"
+              .value=${this.defaultValue}
+            />
+            <span
+              class="${this.defaultValue !== this.midValue ? "active" : ""}"
+              @click=${this.reset}
+              >${reset}</span
+            >
+          </section>
+        </div>
+        <div class="range-container">
+          <input
+            type="range"
+            @input=${this.changeValue}
+            class="slider"
+            min=${this.minValue}
+            max=${this.maxValue}
+            step=${this.step}
+            .value=${this.defaultValue}
+            list="markers"
+          />
+          <div class="range-slider-options">
+            ${this.tickMarksPositions.map(() => html`<span></span>`)}
+          </div>
+          <datalist id="markers">
+            ${this.tickMarksPositions.map(
+              (pos) => html`<option value="${pos}"></option>`
+            )}
+          </datalist>
+        </div>
+        <div class="slider-name">${this.name}</div>
+      </div>
+    `;
+  }
+
+  changeValue(e) {
+    const defaultValue = e.target.value;
+    const isValid = e.target.reportValidity();
+    if (isValid) {
+      this.defaultValue = defaultValue;
+    } else {
+      e.target.setAttribute("aria-invalid", !isValid);
+      this.defaultValue = this.midValue;
+    }
+    this.onChangeCallback(defaultValue);
+  }
+
+  reset() {
+    this.defaultValue = this.midValue;
+    this.onChangeCallback(this.defaultValue);
+  }
+
+  get midValue() {
```

Comment:
"midValue" is not a useful concept. We have an explicit defaultValue, and if it's not given, perhaps it should fall back to the minValue.

Reply:
The naming is misleading here: midValue is used to check against the current value of the slider. 
I can rename midValue to defaultValue 
and defaultValue to currentValue?

Reply:
It is literally calculating the middle between min and max in the line below.

Reply:
Yes, this function is pointless. I meant that it does serve a purpose. But anyways, you can see how it should work after the last commit.